// response-schema/build/src/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.response = void 0;
var codes_1 = require("./codes");
function response(res, options) {
    var code;
    var status;
    var message;
    if (typeof options.code == 'string') {
        code = codes_1.codes[options.code];
    }
    else {
        code = options.code;
    }
    if (typeof options.status === 'undefined') {
        if (code < 400) {
            status = true;
        }
        else {
            status = false;
        }
    }
    else {
        status = options.status;
    }
    ;
    if (options.msg) {
        if (res.locals.i18n) {
            if (res.locals.i18n.messages[options.msg]) {
                message = res.locals.i18n.messages[options.msg];
            }
            else {
                message = options.msg;
            }
        }
    }
    res.status(code).json({
        status: status,
        code: code,
        message: message,
        version: options.version,
        redirect: options.redirect,
        data: options.data
    });
}
exports.response = response;
